src/memory.cpp

```diff
@@ -12,17 +12,31 @@ static unsigned int ulog2(unsigned int numBlocks) {
   return (unsigned int) ceil(log2(std::max(numBlocks, (unsigned int) 2)));
 }
 
-Memory* Memory::create(unsigned int numBlocks, MemEncoding encoding) {
+static unsigned int norm2(unsigned int number) {
+  return (unsigned int) pow(2, ulog2(number));
+}
+
+Memory* Memory::create(
+    unsigned int numGlobalBlocks, unsigned int numLocalBlocks,
+    MemEncoding encoding) {
   switch(encoding) {
   case MemEncoding::SINGLE_ARRAY:
-    return new SingleArrayMemory(numBlocks);
+    return new SingleArrayMemory(norm2(numGlobalBlocks), norm2(numLocalBlocks));
```

Comment:
Why do we want to normalize it?
It will increase the number of blocks, no?

Reply:
Well if we separate `isLocal` bit from bids, then there's no need to normalize it. I'll remove this